scripts/travis_doxygen.sh

```diff
@@ -0,0 +1,22 @@
+#!/bin/bash
+
+set -e
+
+script_folder=`dirname $0`
+pip install --user unidiff
+
+doxygen --version &> /dev/null
+if [[ $? -ne 0 ]]
+then
+  echo "doxygen not installed"
+  echo "Ensure doxygen is installed, then run again"
+  exit 1
+fi
+
+cd src && doxygen &> ../doxygen.log && cd ..
+
+if [ "$TRAVIS_PULL_REQUEST" == "false" ]; then
+  $script_folder/run_diff.sh 'cat doxygen.log' HEAD~1 # Check for errors introduced in last commit
```

Comment:
Passing commands will always (rightfully) raise eyebrows. Wouldn't it be possible to add some detection logic to the script that figures out what to do with a "doxygen.log" file?